packages/neo-one-smart-contract-lib/src/ownership/Ownable.ts

```diff
@@ -0,0 +1,57 @@
+import { Address, SmartContract, createEventNotifier } from '@neo-one/smart-contract';
+
+const notifyTransferOwnership = createEventNotifier<Address | undefined, Address>('transfer_ownership', 'from', 'to');
+
+export function Ownable<TBase extends Constructor<SmartContract>>(base: TBase) {
+  abstract class OwnableClass extends base {
+    protected abstract initialOwner: Address;
+    private mutableOwner: Address | undefined = undefined;
+    private mutableOwnerInitialized = false;
+
+    public get owner(): Address | undefined {
+      return this.mutableOwner !== undefined ? this.mutableOwner : this.initializeOwner();
+    }
+
+    public renounceOwnership() {
+      this.onlyOwner();
+
+      this.mutableOwner = undefined;
+    }
+
+    public transferOwnership(to: Address): boolean {
+      this.onlyOwner();
+
+      notifyTransferOwnership(this.mutableOwner, to);
+      this.mutableOwner = to;
+
+      return true;
+    }
+
+    protected ownerOrThrow(): Address {
+      const owner = this.owner;
+
+      if (owner === undefined) {
+        throw new Error('no owner');
+      }
+
+      return owner;
+    }
+
+    protected onlyOwner() {
```

Comment:
maybe `checkOwner` instead? :thinking: whatever you think. Although I'm not sure the pattern of throwing errors like this is what we want. I think functions should be return false instead, no?

Reply:
Based off: 
https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/ownership/Ownable.sol